processor/servicegraphprocessor/internal/store/store_test.go

```diff
@@ -136,7 +136,8 @@ func TestStoreExpire(t *testing.T) {
 
 	s.Expire()
 	assert.Equal(t, 0, s.len())
-	assert.Equal(t, 0, onCompletedCount)
+	// callball onComplete before edge expired.
```

Comment:
I believe a new test is deserved here, to exercise the new conditions. Can you add a test that will yield true when  trySpeculateEvictHead is called, and one which will yield false?

Reply:
👌